src/lib/salad/hll.c

```diff
@@ -0,0 +1,223 @@
+#include "hll.h"
+#include "hll_empirical.h"
+
+#include <trivia/util.h>
+
+#include <stddef.h>
+#include <math.h>
+#include <assert.h>
+
+/* 6 bits are needed to store the number of leading zeros of 64 bit hash */
+#define HLL_REGISTER_BITS  6
+
+static inline size_t
+hll_n_registers(uint8_t precision)
+{
+	return 1U << precision;
+}
+
+static inline int
+hll_is_valid_cache(double cached)
+{
+	return cached > 0;
+}
+
+static inline void
+hll_invalidate_cache(struct hll *hll)
+{
+	hll->cached_estimation = -1.f;
```

Comment:
I would rather set it to zero.

Reply:
Zero is a valid value for estimation.

Reply:
Then why does `hll_is_valid_cache` consider `hll->cached_estimation == 0` invalid?

Reply:
> Then why does `hll_is_valid_cache` consider `hll->cached_estimation == 0` invalid?

Zero should be considered as a valid value. I'l fix 'hll_is_valid_cache'.